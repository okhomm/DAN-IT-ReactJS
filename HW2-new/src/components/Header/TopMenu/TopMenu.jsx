

const TopMenu = () => {
  return (
    <ul>
      <li>Shop</li>
      <li>Man</li>
      <li>Woman</li>
      <li>Combos</li>
      <li>Joggers</li>
    </ul>
  );
};

export default TopMenu;