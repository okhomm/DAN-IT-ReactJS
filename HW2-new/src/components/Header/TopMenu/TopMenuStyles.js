import styled from 'styled-components';

export const ItemList = styled.ul`
  list-style: none;
`;

export const StyledTopMenuItem = styled.li`
display: inline-block;
padding-right: 40px;

& li:last-child {
padding-right: 0;
}
`;

export const StyledTopMenuLink = styled.a`
text-decoration: none;
color: ${({ theme }) => theme.colorStyles.secondaryColor};

font-family: Causten;
font-size: 22px;
font-style: normal;
font-weight: 700;
line-height: normal;


& a:hover {
color: ${({ theme }) => theme.colorStyles.primaryColor};
}
`;