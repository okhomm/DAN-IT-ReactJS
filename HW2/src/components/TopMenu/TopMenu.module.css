.bg {
    height: 60px;
    background-color: #282828;
    color: white;
    font-family: 'Roboto', sans-serif;

}

.container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: auto;
    max-width: 1200px;
}

.leftBox {
    display: flex;
}

.rightBox {
    display: flex;
}



.catalog {
    display: flex;
    align-items: center;
    min-width: 230px;
    background-color: #F4CF87;
    padding: 18px;
    cursor: pointer;
}

.catalogText {
    font-family: 'Roboto', sans-serif;
    font-style: normal;
    /*font-weight: 600;*/
    font-size: 20px;
    /*line-height: 27px;*/
    color: #282828;
}

.catalogIcon {
    font-size: 24px;
    color: #282828;
    padding-right: 18px;
}

.item {
    display: flex;
    align-items: center;
    padding: 0 30px;
    color: #ffffff;
    font-size: 19px;
    cursor: pointer;
}

.itemPhone{
    display: flex;
    align-items: center;
    margin-left: 120px;
    font-size: 22px;
    cursor: pointer;
}

.item:hover,
.itemPhone:hover{
    color: #F4CF87;
}

.itemIcon{
    padding-right: 10px;
}