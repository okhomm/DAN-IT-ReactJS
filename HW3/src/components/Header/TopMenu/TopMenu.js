import {AiOutlineMenu} from "react-icons/ai";
import {HiPhone} from "react-icons/hi";
import {FaUser} from "react-icons/fa";
import styled from "./TopMenu.module.css";

const TopMenu = () => {
    return (
        <div className={styled.bg}>
            <div className={styled.container}>
                <div className={styled.leftBox}>
                    <nav className={styled.catalog}>
                        <AiOutlineMenu className={styled.catalogIcon}/>
                        <span className={styled.catalogText}>Catalog</span>
                    </nav>
                    <span className={styled.item}>Information</span>
                    <span className={styled.itemPhone}>
                            <HiPhone className={styled.itemIcon}/>
                            0 800 33-96-91
                         </span>
                </div>
                <div className={styled.rightBox}>
                        <span className={styled.item}>
                            <FaUser className={styled.itemIcon}/>
                            Cabinet
                        </span>
                </div>
            </div>
        </div>
    )
}

export default TopMenu;