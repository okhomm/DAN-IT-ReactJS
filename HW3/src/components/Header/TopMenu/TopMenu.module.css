.bg {
    height: 60px;
    background-color: #282828;
    color: white;
    font-family: 'Roboto', sans-serif;

}

.container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: auto;
    max-width: 1200px;
}

.leftBox {
    display: flex;
}

.leftBox a {
    text-decoration: none;
}

.rightBox {
    display: flex;
}

.catalog {
    display: flex;
    align-items: center;
    min-width: 230px;
    background-color: #F4CF87;
    padding: 18px;
}

.catalogText {
    font-family: 'Roboto', sans-serif;
    font-style: normal;
    font-size: 20px;
    color: #282828;
}

.catalogIcon {
    font-size: 24px;
    color: #282828;
    padding-right: 18px;
}

.item {
    display: flex;
    align-items: center;
    padding: 0 30px;
    color: #ffffff;
    font-size: 19px;
}

.itemPhone{
    display: flex;
    align-items: center;
    margin-left: 120px;
    font-size: 22px;
}

.itemIcon{
    padding-right: 10px;
}