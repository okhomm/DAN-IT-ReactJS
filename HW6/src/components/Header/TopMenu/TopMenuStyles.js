import { Link } from 'react-router-dom';
import styled from 'styled-components';
// import theme from '../../../styles/Theme';


export const ItemList = styled.ul`
  list-style: none;
`;

export const StyledTopMenuItem = styled.li`
  display: inline-block;
  padding-right: 40px;

  & li:last-child {
    padding-right: 0;
  }
`;

export const StyledTopMenuLink = styled(Link)`
  text-decoration: none;
  color: ${({ theme }) => (theme && theme.colorStyles && theme.colorStyles.secondaryColor) || 'defaultFallbackColor'};


  font-family: 'Causten', sans-serif;
  font-size: 22px;
  font-style: normal;
  font-weight: 700;
  line-height: normal;

  & a:hover {
    color: ${({ theme }) => (theme && theme.colorStyles && theme.colorStyles.primaryColor) || 'defaultFallbackHoverColor'};
  }
`;
