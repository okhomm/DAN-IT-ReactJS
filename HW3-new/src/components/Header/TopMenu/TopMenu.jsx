import { ItemList, StyledTopMenuItem, StyledTopMenuLink } from './TopMenuStyles';

const TopMenu = () => {
  return (
    <ItemList>
      <StyledTopMenuItem>
        <StyledTopMenuLink href="/">Shop</StyledTopMenuLink>
      </StyledTopMenuItem>
      <StyledTopMenuItem>
        <StyledTopMenuLink href="/">Man</StyledTopMenuLink>
      </StyledTopMenuItem>
      <StyledTopMenuItem>
        <StyledTopMenuLink href="/">Woman</StyledTopMenuLink>
      </StyledTopMenuItem>
      <StyledTopMenuItem>
        <StyledTopMenuLink href="/">Combos</StyledTopMenuLink>
      </StyledTopMenuItem>
      <StyledTopMenuItem>
        <StyledTopMenuLink href="/">Joggers</StyledTopMenuLink>
      </StyledTopMenuItem>
    </ItemList>
  )
};

export default TopMenu;
