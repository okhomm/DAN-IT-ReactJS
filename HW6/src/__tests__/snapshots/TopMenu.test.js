import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import TopMenu from '../../components/Header/TopMenu'

test('renders TopMenu component', () => {
  const { container } = render(
    <Router>
      <TopMenu />
    </Router>
  );

  expect(container).toMatchSnapshot();
});
