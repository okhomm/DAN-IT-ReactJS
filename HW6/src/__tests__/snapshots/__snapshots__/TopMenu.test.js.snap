// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TopMenu component 1`] = `
<div>
  <ul
    class="sc-bdnyFh eMYlAR"
  >
    <li
      class="sc-gtsqUy fHxyKq"
    >
      <a
        class="sc-dlniIP dzFmRf"
        href="/shop"
      >
        Shop
      </a>
    </li>
    <li
      class="sc-gtsqUy fHxyKq"
    >
      <a
        class="sc-dlniIP dzFmRf"
        href="/shop/man"
      >
        Man
      </a>
    </li>
    <li
      class="sc-gtsqUy fHxyKq"
    >
      <a
        class="sc-dlniIP dzFmRf"
        href="/shop/woman"
      >
        Woman
      </a>
    </li>
    <li
      class="sc-gtsqUy fHxyKq"
    >
      <a
        class="sc-dlniIP dzFmRf"
        href="/shop/combos"
      >
        Combos
      </a>
    </li>
    <li
      class="sc-gtsqUy fHxyKq"
    >
      <a
        class="sc-dlniIP dzFmRf"
        href="/shop/joggers"
      >
        Joggers
      </a>
    </li>
  </ul>
</div>
`;
